Remove user creation and modernize

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request removes user creation from the configuration and modernizes the rest of the configuration.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

User creation should now be managed by [cisagov/cyhy-users-non-admin]. I modernized the repo a bit while I was here just to make it more consistent with how we do similar configurations in COOL projects.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Migrate users to [cisagov/cyhy-users-non-admin] and remove them from the state.
- [x] Destroy this configuration as it is deployed.
- [x] Do a fresh apply to lay down the policies using the new configuration.

[cisagov/cyhy-users-non-admin]: https://github.com/cisagov/cyhy-users-non-admin

mcdonnnj committed:
Rename Terraform files to be consistent

Use the filenames we typically used in similar Terraform
configurations.

mcdonnnj committed:
Remove user creation from the configuration

User creation should instead be handled by
cisagov/cyhy-users-non-admin and once the users are created they can be
added to the group defined in this configuration.

mcdonnnj committed:
Move group membership handling into its own file

mcdonnnj committed:
Modify bucket access policy and move it to its own file

Move the bucket access policy into its own Terraform file. Expand the
existing read/write access to be more comprehensive and include the
ability to delete objects.

mcdonnnj committed:
Move the S3 bucket data object into locals.tf

mcdonnnj committed:
Use IAM policy attachment instead of a group policy

Move the IAM policy creation and switch to an IAM policy instead of a
IAM group policy. This policy is then attached to the group instead.
Also add variables to control the name and description for this IAM
policy.

mcdonnnj committed:
Improve Terraform configuration comments

mcdonnnj committed:
Add a variable to control the IAM group name

This value is appended with "-production" for production workspaces or
the current Terraform workspace name otherwise.

mcdonnnj committed:
Move bucket name definition into a local variable

This keeps it grouped with a similar value.

mcdonnnj committed:
Update the README

Update with the output of terraform-docs and update the example tfvars
to reflect changes made.

mcdonnnj committed:
Add an additional CODEOWNER

The base repository should generally have at least three code owners so
that there are two people to review in addition to the person creating
a pull request.

dav3r commented on locals.tf:
```diff
@@ -5,8 +5,18 @@
+}
+
 locals {
   # This is a goofy but necessary way to determine if
   # terraform.workspace contains the substring "prod"
   production_workspace = replace(terraform.workspace, "prod", "") != terraform.workspace
```
As long as we are cleaning up, let's use a regex here:
```suggestion
  # Determine if this is a Production workspace by checking
  # if terraform.workspace begins with "prod-"
  production_workspace =  length(regexall("^prod-", terraform.workspace)) == 1
```

mcdonnnj replied:
Ah, much nicer. Glad you noticed!

dav3r reviewed (approved):
This looks great to me. 👍 

I also noted one line to be modernized that caught my eye.

mcdonnnj committed:
Update production workspace detection

Use a regex to determine if a production workspace is being used.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Use a workspace dependant policy name

This mirrors the IAM group name and the fact that the S3 bucket is
workspace unique.

mcdonnnj committed:
Update the `findings_data_s3_users_group_name` description

Update the description to better represent how the final group name is
determined.

jsf9k commented on README.md:
```diff
@@ -100,8 +99,11 @@ No modules.
+| findingsbucketfullaccess\_policy\_name | The base name to associate with the IAM policy that allows full access to the findings data S3 bucket.  Note that in production workspaces, '-production' is automatically appended this bucket name.  In non-production workspaces, '-<workspace\_name>' is automatically appended to this bucket name. | `string` | `"FindingsBucketFullAccess"` | no |
 | tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
-| usernames | The usernames associated with the accounts to be created.  The format first.last is recommended. | `list(string)` | n/a | yes |
+| users | A list of the usernames for the users that should be given access to the S3 bucket that manages Assessment findings data. | `list(string)` | n/a | yes |
```
Is it worth mentioning that these users _must_ have already been created via [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin)?

mcdonnnj replied:
This is a good point, but I would rather bake it into the pre-requisites in the README. Thoughts?

dav3r replied:
I'm fine as long as it's mentioned somewhere.

jsf9k replied:
I don't think it hurts to mention it in both places, but I will defer to your judgement as long as it is mentioned somewhere.

mcdonnnj replied:
The COOL -iam configurations only mention it as a pre-requisite in the README (which is what I'm using as a "standard").

mcdonnnj replied:
Please see [733c715](https://github.com/cisagov/findings-data-s3-users/pull/21/commits/733c715116291cc9aea398bada4b3fff2845537a).

mcdonnnj committed:
Specify that all users must already exist

Add user account creation as a pre-requisite in the project's README.

dav3r reviewed (approved):
👍 👍
